model/map/src/main/java/org/keycloak/models/map/user/MapUserAdapter.java

```diff
@@ -49,7 +51,8 @@ public String getId() {
 
     @Override
     public String getUsername() {
-        return entity.getUsername();
+        Boolean usernameCaseSensitiveAttribute = realm.getAttribute(REALM_ATTR_USERNAME_CASE_SENSITIVE, REALM_ATTR_USERNAME_CASE_SENSITIVE_DEFAULT);
+        return usernameCaseSensitiveAttribute ? entity.getUsername() : entity.getUsername().toLowerCase();
```

Comment:
I like the new behavior that when the realm attribute is set to false, the username is returned in a case-insensitive format. This is something that wasn't in the change before, and I believe of the changes to the tests need to be reverted to their original state. 

I'm was actually expecting this conversion to happen on the physical layer (JPA implementation). Maybe @hmlnarik could comment on this.

Reply:
> This is something that wasn't in the change before, and I believe of the changes to the tests need to be reverted to their original state.

Do you mean reverting test changes from https://github.com/keycloak/keycloak/pull/13183?

Reply:
> Do you mean reverting test changes from https://github.com/keycloak/keycloak/pull/13183?

Yes - I assume those will fail anyway due to the username being returned in lowercase as long as the setting is false.

Reply:
I can revert those, np with that. Just a note the tests there were updated mostly to lowered case username so it should pass regardless of the setting.

Reply:
You were write, the changed tests would pass. Still I think was good to reverted the ignore-case tests as you did, as checking the case in the tests instead of ignoring it looks better to me. 

This is now resolved for me.